### PR TITLE
Removed loaded modules logging

### LINKED ISSUE #21978: Remove loggin debug of loaded modules
Hi there,

Would you be willing to remove this debug statement from the __init__.py file? It leads to excessive logging output. In my case, it is printing the name of ~1800 modules and really crowding out more valuable debug statements. Thanks!

https://github.com/matplotlib/matplotlib/blob/98115e9b5134f4c71dc196a56f6ed8df4737174a/lib/matplotlib/__init__.py#L1445

### PR DESCRIPTION
**Pull request by cavesdev**
This is just a one-line change. Closes #21978

### CODE REVIEW

**Comment by cavesdev** [comment]
@tacaswell I added a note in next_api_changes under the removals folder documenting this change. Is it enough?

File: doc/api/next_api_changes/removals/21980-CC.rst
Comment (tacaswell) [inline doc/api/next_api_changes/removals/21980-CC.rst:4]: Maybe something more like "The list of currently loaded modules is no longer logged at the DEBUG level at Matplotlib import time.  If you were relying on this please arrange for your own logging." ?

File: doc/api/next_api_changes/removals/21980-CC.rst
Comment (tacaswell) [inline doc/api/next_api_changes/removals/21980-CC.rst:2]: I think this is going to fail the docs for too many decorator markers.

File: doc/api/next_api_changes/removals/21980-CC.rst
Comment (tacaswell) [reply doc/api/next_api_changes/removals/21980-CC.rst:4]: This is going to be read by people maybe a few years in the future so it needs to have enough context that it makes sense without tracking down the issue that this PR closed.

File: doc/api/next_api_changes/removals/21980-CC.rst
Comment (cavesdev) [reply doc/api/next_api_changes/removals/21980-CC.rst]: I was wondering why the doc building failed. Thanks!

File: doc/api/next_api_changes/removals/21980-CC.rst
Comment (cavesdev) [reply doc/api/next_api_changes/removals/21980-CC.rst]: I think the message is more helpful now.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
Looks good to me @cavesdev .  Now we wait for a second review :)

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @cavesdev! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**
